src/index.ts

```diff
@@ -1,33 +1,63 @@
-/**
- * Welcome to Cloudflare Workers! This is your first worker.
- *
- * - Run `npm run dev` in your terminal to start a development server
- * - Open a browser tab at http://localhost:8787/ to see your worker in action
- * - Run `npm run deploy` to publish your worker
- *
- * Learn more at https://developers.cloudflare.com/workers/
- */
-
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface Env {
-  // Example binding to KV. Learn more at https://developers.cloudflare.com/workers/runtime-apis/kv/
-  // MY_KV_NAMESPACE: KVNamespace;
-  //
-  // Example binding to Durable Object. Learn more at https://developers.cloudflare.com/workers/runtime-apis/durable-objects/
-  // MY_DURABLE_OBJECT: DurableObjectNamespace;
-  //
-  // Example binding to R2. Learn more at https://developers.cloudflare.com/workers/runtime-apis/r2/
-  // MY_BUCKET: R2Bucket;
-  //
-  // Example binding to a Service. Learn more at https://developers.cloudflare.com/workers/runtime-apis/service-bindings/
-  // MY_SERVICE: Fetcher;
-  //
-  // Example binding to a Queue. Learn more at https://developers.cloudflare.com/queues/javascript-apis/
-  // MY_QUEUE: Queue;
-}
-
-export default {
-  async fetch (request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
-    return new Response('Hello World!')
-  }
-}
+import { Env } from "./env";
+import { Hono } from "hono";
+import { Error, ServiceAuthType } from "./types";
+import Services from "./services";
+import { streamResponse } from "./utils";
+
+const app = new Hono<{ Bindings: Env }>();
+app
+  .all("*",
+    async (c) => {
+      const xGatewayService = c.req.raw.headers.get("x-gateway-service");
+      const xGatewayServiceHost = c.req.raw.headers.get("x-gateway-service-host");
+      const xGatewayServiceAuthType = c.req.raw.headers.get("x-gateway-service-auth-type");
+      const xGatewayServiceAuthKey = c.req.raw.headers.get("x-gateway-service-auth-key");
+      const xGatewayServiceToken = c.req.raw.headers.get("x-gateway-service-token");
+
+      if (!xGatewayService) {
```

Comment:
I'd still like this to be functional if the service isn't in services.ts - it would then just require the user to specify a key etc

Reply:
Yeah, it is doable, but then the client has to provide the required headers for the service to respond correctly.